src/cursor/cur_std.c

```diff
@@ -1290,6 +1290,38 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
     API_END_RET_STAT(session, ret, cursor_bound);
 }
 
+/*
+ * __wt_cursor_bounds_save --
+ *     Save cursor bounds to restore the state.
+ */
+void
+__wt_cursor_bounds_save(WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *state)
+{
+    WT_ITEM_SET(state->lower_bound, cursor->lower_bound);
+    WT_ITEM_SET(state->upper_bound, cursor->upper_bound);
+    state->flags = cursor->flags;
+}
+
+/*
+ * __wt_cursor_bounds_restore --
+ *     Restore the cursor's bounds state.
+ */
+void
+__wt_cursor_bounds_restore(WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *bounds_state)
+{
+    uint64_t mask;
+
+    mask = WT_CURSTD_BOUND_UPPER | WT_CURSTD_BOUND_UPPER_INCLUSIVE | WT_CURSTD_BOUND_LOWER |
```

Comment:
I would create a new flag called `WT_CURSTD_BOUND_ALL` or `WT_CURSTD_BOUND_MASK` here

Reply:
Agreed we should make a `WT_CURSTD_BOUND_ALL` and put it in cursor.h or similar. We also want to mask the flags in the `state_save` call.

Reply:
[3ae3da2](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/3ae3da25e2566a302ea82fce4a8d45b87daa8bcf)